packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -256,27 +274,23 @@ class _LayoutBuilderElement<ConstraintType extends Constraints> extends RenderOb
 
   @override
   void removeRenderObjectChild(RenderObject child, Object? slot) {
-    final RenderConstrainedLayoutBuilder<ConstraintType, RenderObject> renderObject =
-        this.renderObject;
+    final RenderConstrainedLayoutBuilder<LayoutInfo, RenderObject> renderObject = this.renderObject;
     assert(renderObject.child == child);
     renderObject.child = null;
     assert(renderObject == this.renderObject);
   }
 }
 
-/// Generic mixin for [RenderObject]s created by [ConstrainedLayoutBuilder].
+/// Generic mixin for [RenderObject]s created by [AbstractLayoutBuilder].
 ///
 /// Provides a callback that should be called at layout time, typically in
 /// [RenderObject.performLayout].
-mixin RenderConstrainedLayoutBuilder<
-  ConstraintType extends Constraints,
-  ChildType extends RenderObject
->
+mixin RenderConstrainedLayoutBuilder<LayoutInfo, ChildType extends RenderObject>
     on RenderObjectWithChildMixin<ChildType> {
-  LayoutCallback<ConstraintType>? _callback;
+  void Function(Constraints)? _callback;
```

Comment:
Here and below, should these take generic `LayoutInfo` instead?

Reply:
This currently matches the signature of the callback `RenderObject.invokeLayoutCallback` takes. Made `_updateCallback` private.